File: core/src/com/mygdx/game/Items/Bullet.java
package com.mygdx.game.Items;

import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.math.MathUtils;
import com.badlogic.gdx.math.Vector2;
import com.mygdx.game.Entity.MovableEntity;

/**
 * Created by Adrian on 2017-04-08.
 */

public class Bullet extends MovableEntity {
    private float alpha = 0.3f;
    private int power;

    public Bullet(float x, float y, float rotation) {
        super("bullet", x, y);
        SPEED = 800;
        width = tex.getWidth();
        height = tex.getHeight();
        sprite.setSize(width, height);
        sprite.setOriginCenter();
        sprite.setRotation(rotation);
    }

    public void update(float deltaTime, Bullet b) {
        b.forward(deltaTime);
        alpha += deltaTime;
    }

    @Override
    public void draw(SpriteBatch batch) {
        sprite.setPosition(position.x, position.y);
        sprite.draw(batch, alpha);
    }

    public int getPower() {
        return power;
    }

    public void setPower(int weaponPower) {
        power = weaponPower;
    }

    public void dispose () {
        tex.dispose();
    }
}

File: core/src/com/mygdx/game/Items/Bullet.kt
package com.mygdx.game.Items

import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.mygdx.game.Entity.MovableEntity

/**
 * Created by Adrian on 2017-04-08.
 */

class Bullet(x: Float, y: Float, rotation: Float) : MovableEntity("bullet", x, y) {
    private var bulletAlpha = 0.3f
    var power: Int = 0

    init {
        SPEED = 800f
        entityWidth = tex.width.toFloat()
        entityHeight = tex.height.toFloat()
        sprite.setSize(entityWidth, entityHeight)
        sprite.setOriginCenter()
        sprite.rotation = rotation
    }

    fun update(deltaTime: Float, b: Bullet) {
        b.forward(deltaTime)
        bulletAlpha += deltaTime
    }

    override fun draw(batch: SpriteBatch) {
        sprite.setPosition(position.x, position.y)
        sprite.draw(batch, bulletAlpha)
    }

    override fun dispose() {
        tex.dispose()
    }
}
